### PR TITLE
SLE-923,SLE-927: Create ITs and incorporate validation feedback

### PR DESCRIPTION
**Pull request by thahnen**
## Changes to the `dev` target platform

In order to run RedDeer tests from the latest Eclipse IDE (that is running on top of Java 21), we also now have to use the Java 21-based Eclipse IDE as a reference in our development environment. I didn't notice it before as I developed against a dogfooding version of the Eclipse IDE that was powered by my locally installed Java 17 runtime.

## SLE-927: Validation feedback

Feedback from the PM was incorporated before implementing the ITs. To make it clearer, what was meant with button descriptions and notification titles displayed on failure. See the commit messages for more information.

## SLE-923: Integration tests

A set of integration tests covering most of the cases by emulating a call from SonarCloud to the internal endpoint of SonarLint. This includes the default `file not found` as well as canceling, declining, and applying changes as well as already implemented fixes - therefore, the suggestion could not be found.

### CODE REVIEW

File: its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java [hunk 1]
```diff
@@ -19,38 +19,67 @@
...
+import static org.assertj.core.api.Assertions.fail;
 
 public class SonarCloudConnectedModeTest extends AbstractSonarLintTest {
   private static final String TIMESTAMP = Long.toString(Instant.now().toEpochMilli());
   private static final String SONARCLOUD_STAGING_URL = "https://sc-staging.io";
   private static final String SONARCLOUD_ORGANIZATION_KEY = "sonarlint-it";
   private static final String SONARCLOUD_USER = "sonarlint-it";
-  private static final String SONARCLOUD_PASSWORD = System.getenv("SONARCLOUD_IT_PASSWORD");
-
+  private static final String SONARCLOUD_PASSWORD = "ERiyhgujD5637FGErerev";// System.getenv("SONARCLOUD_IT_PASSWORD");
```
Comment (sophio-japharidze-sonarsource) [inline its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java:73]: uh-oh 🙈

File: its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java [hunk 2]
```diff
@@ -19,38 +19,67 @@
...
   private static final String TOKEN_NAME = "SLE-IT-" + TIMESTAMP;
-
   private static final String CONNECTION_NAME = "connection";
+  private static final String SAMPLE_JAVA_ISSUES_KEY = "sonarlint-its-sample-java-issues";
```
Comment (sophio-japharidze-sonarsource) [inline its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java:76]: `SAMPLE_JAVA_ISSUES_PROJECT_KEY` I assume

File: its/org.sonarlint.eclipse.its.shared/src/org/sonarlint/eclipse/its/shared/reddeer/dialogs/ConfirmConnectionCreationDialog.java [hunk 3]
```diff
@@ -22,12 +22,17 @@
 import org.eclipse.reddeer.swt.impl.shell.DefaultShell;
 
 public class ConfirmConnectionCreationDialog extends DefaultShell {
-  public ConfirmConnectionCreationDialog() {
-    super("Do you trust this SonarQube server?");
+  private final boolean isSonarCloud;
+
+  public ConfirmConnectionCreationDialog(boolean isSonarCloud) {
+    super(isSonarCloud
+      ? "Do you trust SonarCloud?"
```
Comment (sophio-japharidze-sonarsource) [inline its/org.sonarlint.eclipse.its.shared/src/org/sonarlint/eclipse/its/shared/reddeer/dialogs/ConfirmConnectionCreationDialog.java:29]: "Do you trust this SonarCloud organization?

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/MessageDialogUtils.java [hunk 4]
```diff
@@ -28,29 +28,41 @@
...
+  public static void fileNotFound(String message) {
+    showError("File not found", message);
+  }
+
+  public static void dialogCancelled(String message) {
+    showError("Dialog cancelled", message);
+  }
+
+  public static void branchMismatch(String message) {
+    showError("Branch not matching", message);
```
Comment (sophio-japharidze-sonarsource) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/MessageDialogUtils.java:49]: "branch does not match" for consistency?

**Review by sophio-japharidze-sonarsource — COMMENTED** [review]
Looks great with minor comments;

If possible, maybe we could also add another IT(s) with multiple changes incoming at once, and where one of them is not applicable (outside of file range)

File: its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java (hunk 2 above)
Comment (thahnen) [reply its/org.sonarlint.eclipse.its.connected.sc/src/org/sonarlint/eclipse/its/connected/sc/SonarCloudConnectedModeTest.java]: Fixed

File: its/org.sonarlint.eclipse.its.shared/src/org/sonarlint/eclipse/its/shared/reddeer/dialogs/ConfirmConnectionCreationDialog.java (hunk 3 above)
Comment (thahnen) [reply its/org.sonarlint.eclipse.its.shared/src/org/sonarlint/eclipse/its/shared/reddeer/dialogs/ConfirmConnectionCreationDialog.java]: Changed, yes this is clearer

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/MessageDialogUtils.java (hunk 4 above)
Comment (thahnen) [reply org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/MessageDialogUtils.java]: Changed, yes this is clearer

**Review by sophio-japharidze-sonarsource — APPROVED** [review]

**Pull request merged**